extensions/mssql/src/objectManagement/ui/serverPropertiesDialog.ts

```diff
@@ -0,0 +1,84 @@
+	private memoryInput: azdata.InputBoxComponent;
+	private operatingSystemInput: azdata.InputBoxComponent;
+
+	private memoryTab: azdata.Tab;
+	private memorySection: azdata.GroupContainer;
+	private minServerMemoryInput: azdata.InputBoxComponent;
+	private maxServerMemoryInput: azdata.InputBoxComponent;
+
+	constructor(objectManagementService: IObjectManagementService, options: ObjectManagementDialogOptions) {
+		super(objectManagementService, options);
+	}
+
+	protected override get helpUrl(): string {
+		return ViewServerPropertiesDocUrl;
+	}
+
+	protected async initializeUI(): Promise<void> {
+		this.initializeGeneralSection();
+		this.initializeMemorySection();
+		const serverPropertiesTabGroup = { title: '', tabs: [this.generalTab, this.memoryTab] };
+		const serverPropertiesTabbedPannel = this.modelView.modelBuilder.tabbedPanel().withTabs([serverPropertiesTabGroup]).component();
+		this.formContainer.addItem(serverPropertiesTabbedPannel);
+	}
+
+	private initializeGeneralSection(): void {
+		this.nameInput = this.createInputBox(localizedConstants.NameText, async (newValue) => {
+			this.objectInfo.name = newValue;
+		}, this.objectInfo.name, this.options.isNewObject);
+		const nameContainer = this.createLabelInputContainer(localizedConstants.NameText, this.nameInput);
+
+		this.languageDropdown = this.createDropdown(localizedConstants.LanguageText, undefined, [this.objectInfo.language], this.objectInfo.language, this.options.isNewObject);
+		const languageContainer = this.createLabelInputContainer(localizedConstants.LanguageText, this.languageDropdown);
+
+		this.memoryInput = this.createInputBox(localizedConstants.MemoryText, undefined, this.objectInfo.memoryInMb.toString(), this.options.isNewObject, 'number');
+		const memoryContainer = this.createLabelInputContainer(localizedConstants.MemoryText, this.memoryInput);
+
+		this.operatingSystemInput = this.createInputBox(localizedConstants.OperatingSystemText, undefined, this.objectInfo.operatingSystem, this.options.isNewObject);
+		const operatingSystemContainer = this.createLabelInputContainer(localizedConstants.OperatingSystemText, this.operatingSystemInput);
+
+		this.generalSection = this.createGroup('', [
+			nameContainer,
+			languageContainer,
+			memoryContainer,
+			operatingSystemContainer
+		], false);
+
+		this.generalTab = this.createTab('generalId', localizedConstants.GeneralSectionHeader, this.generalSection);
+	}
+
+	private initializeMemorySection(): void {
+		this.minServerMemoryInput = this.createInputBox(localizedConstants.minServerMemoryText, async (newValue) => {
+			this.objectInfo.minMemoryInMb = +newValue;
+		}, this.objectInfo.minMemoryInMb.toString(), true, 'number');
+		const minMemoryContainer = this.createLabelInputContainer(localizedConstants.minServerMemoryText, this.minServerMemoryInput);
+
+		this.maxServerMemoryInput = this.createInputBox(localizedConstants.maxServerMemoryText, async (newValue) => {
+			this.objectInfo.maxMemoryInMb = +newValue;
```

Comment:
what does `+newValue` mean?

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Unary_plus

Converts the value to a number (which we already guarantee by having the input box type be number)